src/app/board/board.component.ts

```diff
@@ -15,57 +12,36 @@ import { ColumnService } from './column.service';
   templateUrl: './board.component.html',
   styleUrls: ['./board.component.less'],
 })
-export class BoardComponent implements OnInit {
-  columns: Column[] = [];
-  cards: Card[] = [];
+export class BoardComponent {
+  columns$: Observable<Column[]>;
 
   @HostBinding('class')
   private hostClass = 'mat-app-background';
 
   constructor(
     private columnService: ColumnService,
     private cardService: CardService,
-  ) {}
-
-  ngOnInit() {
-    this.populate();
-  }
-
-  populate() {
-    combineLatest(
-      this.columnService.getColumns(),
-      this.cardService.getCards(),
-    ).subscribe(([columns, cards]) => {
-      this.columns = columns;
-      this.cards = cards;
-    });
+  ) {
+    this.columns$ = this.columnService.columns$;
   }
 
   getColumnNodeId(column: Column): string {
     return `column-${column.id}`;
   }
 
-  getCardsOfColumn(column: Column): Card[] {
-    return column.cards
-      .map(id => this.cards.find(card => card.id === id))
-      .filter(Boolean);
+  getCardsOfColumn(column: Column): Observable<Card[]> {
+    return combineLatest(
```

Comment:
Podemos usar o `of` do `Observable` aqui por que o `combineLatest` a gente usa em situações em que precisamos juntar a emissão de varias fontes. Ja utilizei esse método para combinar `Store.select` com `router.data`. De uma olhada na documentação do:

* [`of`](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#static-method-of)
* [`combineLatest`](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#static-method-combineLatest)

Um detalhe importante aqui: toda vez que o Angular renderizar o componente um novo `Observable` sera criado, mesmo os objetos ainda serão os mesmos. Isso acontece toda vez que o Angular precisa fazer alguma coisa nesse componente ou ate msm toda vez que o Angular executar o Change Detection na aplicação. Experimente colocar aqui um `console.log` para vc ver quantas vezes esse método e executado.

Uma opção que podemos experimentar usar ao invés de um método eh criarmos um [pure `Pipe`](https://angular.io/guide/pipes). Um pure pipe do Angular somente vai ser executado novamente se a referencia do objeto passado mudar. Isso pode melhorar significantemente a performance do app. Então, nesse pipe voce pode migrar a mesma lógica para filtrar os cards.

Um outro detalhe. Acredito que exista um possível problema de memory leak com o `combineLatest` dessa forma. Pelo que entendo do conceito ele não completa o `Observable` após a 1a emissão de valor (aqui um `Card[]`.

PS: Nao consegui entender a motivação pra mudar o retorno de `Card[]` para `Observable<Card[]>`.

Reply:
> PS: Nao consegui entender a motivação pra mudar o retorno de `Card[]` para `Observable<Card[]>`.

Então, minha ideia foi remover a necessidade do componente pedir ao Service para buscar os dados (antigo método `populate` desse componente). Pra isso eu combinei o subscription de tanto as colunas quanto aos cards com a checagem da necessidade de dar um fetch.

`this.cardService.getCard(id)` agora retorna um `Observable<Card>`, por isso usei o `combineLatest` também.

Mas eu concordo com seus comentários relacionados à performance. Eu garanti que apenas um request para o cards aconteceria, mesmo com múltiplos subscribers. Agora vou checar a qtd de observables que estão sendo criados e a frequência.

Reply:
Principalmente por causa do drag-and-drop esse metodo era realmente chamado muitas vezes.

Gostei do fato que os pipes sao considerados puros por padrao :)

Corrigido na nova branch: https://github.com/DiogoDoreto/ng-kanban/commit/04e7903a101b1ae23091dcc910b397648761bc61